### PR TITLE
refactor: simplify the Lifecycle Watcher

### PR DESCRIPTION
**Pull request by Lightning00Blade**
We can expect the correct order of event for `DOMContentLoaded` and `load` events,
that the main frame's will come last [upstream CL](https://chromium-review.googlesource.com/c/chromium/src/+/4684286).

### CODE REVIEW

File: packages/puppeteer-core/src/cdp/LifecycleWatcher.ts [hunk 1]
```diff
@@ -227,41 +223,19 @@ export class LifecycleWatcher {
...
-      for (const event of expectedLifecycle) {
-        if (!frame._lifecycleEvents.has(event)) {
-          return false;
-        }
-      }
-      // TODO(#1): Its possible we don't need this check
-      // CDP provided the correct order for Loading Events
-      // And NetworkIdle is a global state
-      // Consider removing
-      for (const child of frame.childFrames()) {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/LifecycleWatcher.ts:255]: so was it here only for the empty frames? isn't this code also checks for networkidle events which could come in any order of the frames/subframes? Is it definitely not possible that the load or domcontent loader events arrive after the main frame load/domcontentloaded?

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/cdp/LifecycleWatcher.ts:255]: So for the Lifecycle events they now (after the linked CL) come in order, that's only true for parsing so anything coming from the HTML or Scripts (evaluateOnNewDocument included) will be correct.
As for the `NetworkIdle` I have to double check but I think the state for every frame is track on the same NetworkIdle detector so it's global for the page (comment also suggest that).

Comment (OrKoN) [reply packages/puppeteer-core/src/cdp/LifecycleWatcher.ts:255]: I am not sure it can be tracked globally with OOPIFs

Comment (Lightning00Blade) [reply packages/puppeteer-core/src/cdp/LifecycleWatcher.ts:255]: Yeah you may be correct. I don't think we have such test. I will write one here.

**Review by Leezsusan — COMMENTED** [review]
Hey man don't be reviewing my darn life cycle I came before not after. Time is different here you know

****

File: packages/puppeteer-core/src/cdp/LifecycleWatcher.ts (hunk 1 above)
Comment (OrKoN) [reply packages/puppeteer-core/src/cdp/LifecycleWatcher.ts:255]: @Lightning00Blade do you want to keep the PR open?

**Comment by Lightning00Blade** [comment]
Closing as the network idle parts need this to properly detect that part of the lifecycle.

**Pull request closed without merging**